src/Uno.UI/UI/Xaml/Window.Android.cs

```diff
@@ -95,12 +101,37 @@ private int GetLogicalStatusBarHeight()
 				int resourceId = Android.Content.Res.Resources.System.GetIdentifier("status_bar_height", "dimen", "android");
 				if (resourceId > 0)
 				{
-					logicalStatusBarHeight = (int)(Android.Content.Res.Resources.System.GetDimensionPixelSize(resourceId) / Android.App.Application.Context.Resources.DisplayMetrics.Density);
+					logicalStatusBarHeight = ViewHelper.PhysicalToLogicalPixels(Android.Content.Res.Resources.System.GetDimensionPixelSize(resourceId));
 				}
 			}
 
 			return logicalStatusBarHeight;
 		}
+
+		private double GetLogicalNavigationBarHeight()
+		{
+			var logicalNavigationBarHeight = 0d;
+			var metrics = new DisplayMetrics();
+			var defaultDisplay = (ContextHelper.Current as Activity)?.WindowManager?.DefaultDisplay;
+
+			var activity = ContextHelper.Current as Activity;
+			var decorView = activity.Window.DecorView;
+
+			if (defaultDisplay != null && IsNavigationBarVisible)
+			{
+				defaultDisplay.GetMetrics(metrics);
+				var usableHeight = metrics.HeightPixels;
+
+				defaultDisplay.GetRealMetrics(metrics);
+				var realHeight = metrics.HeightPixels;
+
+				logicalNavigationBarHeight = realHeight > usableHeight
+					? ViewHelper.PhysicalToLogicalPixels(realHeight - usableHeight)
```

Comment:
so neither GetMetrics nor GetRealMetrics include the status bar?

Reply:
Need to confirm this one but I think neither or both are handling the status bar. On it

Reply:
It looks like the status bar is always included in both.
See this : https://medium.com/@elye.project/an-important-note-when-managing-different-screen-height-3140e26e381a

Reply:
According to my tests, it is included only if the status bar is translucent, it is not otherwise.
Should be the same for the navigation bar (on it)